core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/columns/ColumnsResolver.kt

```diff
@@ -0,0 +1,40 @@
+package org.jetbrains.kotlinx.dataframe.columns
+
+import org.jetbrains.kotlinx.dataframe.ColumnsSelector
+import org.jetbrains.kotlinx.dataframe.DataColumn
+import org.jetbrains.kotlinx.dataframe.DataFrame
+import org.jetbrains.kotlinx.dataframe.impl.columns.*
+
+/**
+ * ## ColumnsResolver
+ * Entity that can be resolved into a list of [columns][DataColumn].
+ *
+ * Used as a return type of [ColumnsSelector].
+ *
+ * Implemented by [SingleColumn] and [ColumnSet].
+ *
+ * @param C common type of resolved columns
+ * @see [SingleColumn]
+ * @see [ColumnSet]
+ * @see [TransformableColumnSet]
+ * @see [TransformableSingleColumn]
+ */
+public interface ColumnsResolver<out C> {
```

Comment:
If you intend for it to be implemented by two classes, might consider making it sealed interface

Reply:
![image](https://github.com/Kotlin/dataframe/assets/17594275/439f3939-7c69-48d4-b46c-c0121d1963f6)
I would, but that would impact the flexibility of the type. The original `ColumnSet` also wasn't sealed. I think the `ColumnsResolver` interface is simple enough to be reused in other parts of the API eventually.